File: 0560-Subarray Sum Equals K/0560-Subarray Sum Equals K.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_map<int, int> table;
        int sum = 0;
        table[0] = 1;
        int count = 0;
        for (int num : nums)
        {
            sum += num;
            count += table[sum - k];
            ++table[sum];
        }
        
        return count;
    }
};

File: 0501-0600/0560-Subarray Sum Equals K/0560-Subarray Sum Equals K.py
import collections
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        presum = count = 0
        table = collections.Counter([0])
        for num in nums:
            presum += num
            count += table[presum - k]
            table[presum] += 1
        return count
